crates/turbopack-ecmascript-plugins/src/transform/swc_ecma_transform_plugins.rs

```diff
@@ -0,0 +1,223 @@
+/// compiled, serialized wasmer::Module instead of raw file bytes to reduce the
+/// cost of the compilation.
+#[turbo_tasks::value(
+    transparent,
+    serialization = "none",
+    eq = "manual",
+    into = "new",
+    cell = "new"
+)]
+pub struct SwcPluginModule(
+    #[turbo_tasks(trace_ignore)]
+    #[cfg(feature = "swc_ecma_transform_plugin")]
+    pub swc_core::plugin_runner::plugin_module_bytes::CompiledPluginModuleBytes,
+    // Dummy field to avoid turbo_tasks macro complains about empty struct.
+    // This is due to we can't import CompiledPluginModuleBytes by default, it should be only
+    // available for the target / platforms can support swc plugins (which can build wasmer)
+    #[cfg(not(feature = "swc_ecma_transform_plugin"))] pub (),
+);
+
+impl SwcPluginModule {
+    pub fn new(plugin_name: &str, plugin_bytes: Vec<u8>) -> Self {
+        Self {
+            #[cfg(feature = "swc_ecma_transform_plugin")]
+            0: {
+                use swc_core::plugin_runner::plugin_module_bytes::{
+                    CompiledPluginModuleBytes, RawPluginModuleBytes,
+                };
+                CompiledPluginModuleBytes::from(RawPluginModuleBytes::new(
+                    plugin_name.to_string(),
+                    plugin_bytes,
+                ))
+            },
+            #[cfg(not(feature = "swc_ecma_transform_plugin"))]
+            0: (),
+        }
+    }
+}
+
+#[turbo_tasks::value(shared)]
+struct UnsupportedSwcEcmaTransformPluginsIssue {
+    pub context: FileSystemPathVc,
+}
+
+#[turbo_tasks::value_impl]
+impl Issue for UnsupportedSwcEcmaTransformPluginsIssue {
+    #[turbo_tasks::function]
+    fn severity(&self) -> IssueSeverityVc {
+        IssueSeverity::Warning.into()
+    }
+
+    #[turbo_tasks::function]
+    fn category(&self) -> StringVc {
+        StringVc::cell("transform".to_string())
+    }
+
+    #[turbo_tasks::function]
+    async fn title(&self) -> Result<StringVc> {
+        Ok(StringVc::cell(format!(
+            "Unsupported SWC Ecma transform plugins on this platform."
+        )))
+    }
+
+    #[turbo_tasks::function]
+    fn context(&self) -> FileSystemPathVc {
+        self.context
+    }
+
+    #[turbo_tasks::function]
+    fn description(&self) -> StringVc {
+        StringVc::cell(
+            "Turbopack does not yet support running SWC ecma transform plugins on this platform."
+                .to_string(),
+        )
+    }
+}
+
+#[derive(Debug)]
+pub struct SwcEcmaTransformPluginsTransformer {
+    #[cfg(feature = "swc_ecma_transform_plugin")]
+    plugins: Vec<(SwcPluginModuleVc, serde_json::Value)>,
+}
+
+impl SwcEcmaTransformPluginsTransformer {
+    #[cfg(feature = "swc_ecma_transform_plugin")]
+    pub fn new(plugins: Vec<(SwcPluginModuleVc, serde_json::Value)>) -> Self {
+        Self { plugins }
+    }
+
+    #[cfg(not(feature = "swc_ecma_transform_plugin"))]
+    pub fn new() -> Self {
+        Self {}
+    }
```

Comment:
Why allow creating this object at all if that feature is not enabled? This whole module could be gated under `#[cfg(feature = "swc_ecma_transform_plugin")]`

Reply:
If this whole module is behind of flag, turbo_tasks's `register` fn panics when build. I tried to make it work but couldn't.

Reply:
Ah yes, I've run into this before. This is because the macro will still pick up the structures. I've created [WEB-1102](https://linear.app/vercel/issue/WEB-1102/cant-use-rust-features-with-turbo-tasks) to track this. Could you add a short comment to refer to this issue?

Reply:
Yup, added.